lib/model.js

```diff
@@ -2108,29 +2109,33 @@ Model.insertMany = function(arr, options, callback) {
   }
 
   var toExecute = [];
-  arr.forEach(function(doc) {
+  var validationError = new ValidationError(); // Group all ValidationErrors together
+  arr.forEach(function(doc, index) {
     toExecute.push(function(callback) {
       doc = new _this(doc);
       doc.validate({ __noPromise: true }, function(error) {
         if (error) {
-          // Option `ordered` signals that insert should be continued after reaching
-          // a failing insert. Therefore we delegate "null", meaning the validation
-          // failed. It's up to the next function to filter out all failed models
-          if (options != null && typeof options === 'object' && options['ordered'] === false) {
-            return callback(null, null);
-          }
-          return callback(error);
+          validationError.errors[index] = error;
+          // Never return the error, as it will cause parallelLimit to immediately
+          // abort. Return null doc for `ordered: false` option.
+          return callback(null, null);
```

Comment:
If `ordered` isn't set to `true` we still want it to abort and only give a single validation error.